### PR TITLE
[Twilio Adapter] Apply feedback and prep for release

### PR DESCRIPTION
**Pull request by matiasroldan6**
## Description
Applied Gabo's feedback in order to even all the adapters and getting them ready for release.

## Details
- Applied feedback to Twilio Adapter from issue 2542 (MS).
- Refactored adapter to receive wrapper, and wrapper to receive options in turn.
- Adapted unit tests to reflect changes.
- Changed files payload in unit tests.
- Refactored ProcessAsync method to keep consistency with the rest of the adapters.

### CODE REVIEW

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioAdapterTests.cs [hunk 1]
```diff
@@ -16,57 +16,28 @@ namespace Microsoft.Bot.Builder.Adapters.Twilio.Tests
...
-        public void Constructor_Should_Fail_With_Null_AuthToken()
-        {
-            var options = new TwilioAdapterOptions("Test", "Test", null, new Uri("http://contoso.com"));
-
-            Assert.Throws<ArgumentException>(() => { new TwilioAdapter(options, new Mock<TwilioClientWrapper>().Object); });
-        }
-
-        [Fact]
-        public void Constructor_Should_Fail_With_Null_TwilioApi()
+        public void Constructor_Should_Fail_With_Null_TwilioAPIWrapper()
```
Comment (LCarbonellRod) [inline tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioAdapterTests.cs]: Is it okay for the name of the tests to have underscores?

Comment (matiasroldan6) [reply tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioAdapterTests.cs]: true. It's not

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioAdapterTests.cs [hunk 2]
```diff
@@ -16,57 +16,28 @@ namespace Microsoft.Bot.Builder.Adapters.Twilio.Tests
...
         {
             var options = new TwilioAdapterOptions("Test", "Test", "Test", new Uri("http://contoso.com"));
+            TwilioClientWrapper twilioClientWrapper = null;
 
-            Assert.Throws<ArgumentNullException>(() => { new TwilioAdapter(options, null); });
+            Assert.Throws<ArgumentNullException>(() => { new TwilioAdapter(twilioClientWrapper); });
```
[the remark this answers is not in the document]
Comment (LCarbonellRod) [reply tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioAdapterTests.cs]: Ok, I understand, Thanks!

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioClientWrapper.cs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ﻿// Copyright (c) Microsoft Corporation.All rights reserved.
```
Comment (ceciliaavila) [inline libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioClientWrapper.cs]: There's a space missing.
```suggestion
// Copyright (c) Microsoft Corporation. All rights reserved.
```

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioClientWrapper.cs [hunk 4]
```diff
@@ -13,15 +14,33 @@ namespace Microsoft.Bot.Builder.Adapters.Twilio
...
-        public virtual void LogIn(string username, string password)
+        /// <param name="options">An object containing API credentials, a webhook verification token and other options.</param>
+        public TwilioClientWrapper(TwilioAdapterOptions options)
         {
-            TwilioClient.Init(username, password);
+            Options = options ?? throw new ArgumentNullException(nameof(options));
+
+            if (string.IsNullOrWhiteSpace(options.TwilioNumber))
+            {
+                throw new ArgumentException("TwilioNumber is a required part of the configuration.", nameof(options));
```
Comment (ceciliaavila) [inline libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioClientWrapper.cs]: Here, the exception should name the exact parameter as we did on Webex. The same for the rest of the exceptions in this method.
```suggestion
                throw new ArgumentException(nameof(options.TwilioNumber));
```

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioAdapterTests.cs [hunk 5]
```diff
@@ -16,57 +16,28 @@ namespace Microsoft.Bot.Builder.Adapters.Twilio.Tests
...
         }
 
         [Fact]
-        public void Constructor_WithArguments_Succeeds()
+        public void Constructor_With_TwilioAPIWrapper_Succeeds()
         {
             var options = new TwilioAdapterOptions("Test", "Test", "Test", new Uri("http://contoso.com"));
 
-            Assert.NotNull(new TwilioAdapter(options, new Mock<TwilioClientWrapper>().Object));
+            Assert.NotNull(new TwilioAdapter(new Mock<TwilioClientWrapper>(options).Object));
```
Comment (ceciliaavila) [inline tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioAdapterTests.cs]: We should have the options as a property to use them in all tests without duplicating code.
```suggestion
            Assert.NotNull(new TwilioAdapter(new Mock<TwilioClientWrapper>(_testOptions).Object));
```

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioAdapterTests.cs [hunk 6]
```diff
@@ -15,58 +18,26 @@ namespace Microsoft.Bot.Builder.Adapters.Twilio.Tests
 {
     public class TwilioAdapterTests
     {
-        [Fact]
-        public void Constructor_Should_Fail_With_Null_Options()
```
Comment (ceciliaavila) [inline tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioAdapterTests.cs]: Shouldn't be positive tests before negative ones?

**Pull request merged**